packages/devtools_app/lib/src/standalone_ui/ide_shared/property_editor/property_editor_controller.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app_shared/utils.dart';
+import 'package:flutter/foundation.dart';
+
+import '../../../service/editor/api_classes.dart';
+import '../../../service/editor/editor_client.dart';
+
+class PropertyEditorController extends DisposableController
+    with AutoDisposeControllerMixin {
+  PropertyEditorController(this.editorClient) {
+    _init();
+  }
+
+  final EditorClient editorClient;
+
+  TextDocument? _currentDocument;
+  CursorPosition? _currentCursorPosition;
+
+  ValueListenable<List<EditableArgument>> get editableArgs => _editableArgs;
+  final _editableArgs = ListValueNotifier<EditableArgument>([]);
```

Comment:
if we are only ever calling replaceAll, then we should just use a ValueNotifier. ListValueNotifier is the right choice when we want to modify the list and notify listeners, but if we are replacing all then this is the same as calling valueNotifier.value = newList

Reply:
good to know, thanks! switched to a normal value notifier